.github/workflows/deploy-pip-poetry.yml

```
@@ -0,0 +1,68 @@
+name: Deploy
+
+on:
+
+  workflow_call:
+    inputs:
+        os:
+          required: true
+          type: string
+        python-version:
+          required: true
+          type: string
+        publish:
+          required: false
+          type: boolean
+          default: false
+    secrets:
+      PYPI-TOKEN:
+        required: false
+
+
+jobs:
+  build:
+    runs-on: ${{ inputs.os }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v2
```

```suggestion
        uses: actions/setup-python@v4
```
https://github.com/actions/setup-python